### PR TITLE
Updated to honor no_proxy environment variable as well

### PR DESCRIPTION
**Pull request by BobSilent**
When using http_proxy environment variable for setting the proxy (see #2698), then also no_proxy environment variable should be considered for setting the correct bypass list.

### CODE REVIEW

File: src/Microsoft.VisualStudio.Services.Agent/VstsAgentWebProxy.cs [hunk 1]
```diff
@@ -130,20 +130,26 @@ public void LoadProxyBypassList()
             if (File.Exists(proxyBypassFile))
             {
                 Trace.Verbose($"Try read proxy bypass list from file: {proxyBypassFile}.");
-                foreach (string bypass in File.ReadAllLines(proxyBypassFile))
+                foreach (string bypass in File.ReadAllLines(proxyBypassFile).Where(value => !string.IsNullOrEmpty(value)).Select(value => value.Trim()))
```
Comment (jtpetty) [inline src/Microsoft.VisualStudio.Services.Agent/VstsAgentWebProxy.cs]: this should probably be IsNullOrWhitespace

File: src/Microsoft.VisualStudio.Services.Agent/VstsAgentWebProxy.cs [hunk 2]
```diff
@@ -158,7 +164,7 @@ private void LoadProxySetting()
 
             if (string.IsNullOrEmpty(ProxyAddress))
             {
-                foreach (var envVar in new string[] {"VSTS_HTTP_PROXY", "http_proxy"})
+                foreach (var envVar in new string[] { "VSTS_HTTP_PROXY", "HTTP_PROXY" })
```
Comment (jtpetty) [inline src/Microsoft.VisualStudio.Services.Agent/VstsAgentWebProxy.cs]: this needs to stay "http_proxy". Environment variables on unix are case sensitive and the convention for proxy related variables it to be lowercase. Same for "no_proxy" above.

Comment (jtpetty) [reply src/Microsoft.VisualStudio.Services.Agent/VstsAgentWebProxy.cs]: Actually, a little more research shows "NO_PROXY" should be fine, but "http_proxy" needs to stay lowercase.

https://unix.stackexchange.com/questions/212894/whats-the-right-format-for-the-http-proxy-environment-variable-caps-or-no-ca

Comment (BobSilent) [reply src/Microsoft.VisualStudio.Services.Agent/VstsAgentWebProxy.cs]: I will do it all lower-cas. I also saw a description here
https://www.gnu.org/software/wget/manual/html_node/Proxies.html

**Comment by pinkfloydx33** [comment]
The proxy bypass list is expected to be a list of regular expressions (`SDK.AgentWebProxy` parses them as such) but the `no_proxy` environment variable is just a list of hostname/suffix-patterns. If you're using `no_proxy` for other processes, you can't fill it in with Regular Expressions and in many cases the values in `no_proxy` will be interpreted _incorrectly_ as valid regexes. 

For example, "10.1.*"  is a valid regex, but its regex interpretation does _not_ jive with it's representation as a hostname/suffix:  `10`, followed by any char, followed by `1`, followed by any char(0+ times) is not the same as "the network beginning 10.1."

**Pull request merged**